toolchain/check/member_access.cpp

```diff
@@ -207,24 +217,26 @@ auto PerformMemberAccess(Context& context, Parse::MemberAccessExprId node_id,
                         SemIR::TypeId, SemIR::NameId);
       context.emitter().Emit(node_id, QualifiedExprNameNotFound, base_type_id,
                              name_id);
-      break;
+      return SemIR::InstId::BuiltinError;
     }
-    // TODO: `ConstType` should support member access just like the
-    // corresponding non-const type, except that the result should have `const`
-    // type if it creates a reference expression performing field access.
-    default: {
-      if (base_type_id != SemIR::TypeId::Error) {
-        CARBON_DIAGNOSTIC(QualifiedExprUnsupported, Error,
-                          "Type `{0}` does not support qualified expressions.",
-                          SemIR::TypeId);
-        context.emitter().Emit(node_id, QualifiedExprUnsupported, base_type_id);
-      }
-      break;
+
+    if (base_type_id != SemIR::TypeId::Error) {
+      CARBON_DIAGNOSTIC(QualifiedExprUnsupported, Error,
+                        "Type `{0}` does not support qualified expressions.",
+                        SemIR::TypeId);
+      context.emitter().Emit(node_id, QualifiedExprUnsupported, base_type_id);
     }
+    return SemIR::InstId::BuiltinError;
   }
 
-  // Should only be reached on error.
-  return SemIR::InstId::BuiltinError;
+  // Perform lookup into the base type.
+  auto member_id = LookupMemberNameInScope(context, node_id, base_id, name_id,
+                                           *name_scope_id);
+
+  // Perform instance binding if we found an instance member.
+  member_id = PerformInstanceBinding(context, node_id, base_id, member_id);
+
+  return member_id;
```

Comment:
FWIW this might be nested less if you switch the `if(!` above; you could also put `name_scope_id` as a condition-scoped variable.

Reply:
I think it's easier to follow if the error case is the early exit, rather than the success case.